### ISSUE TITLE
Support for auto_mitigate in the resource azurerm_monitor_scheduled_query_rules_alert

### ISSUE THREAD

**Issue by JamesDLD** [hashicorp/terraform-provider-azurerm#12691]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Support for auto_mitigate in the resource azurerm_monitor_scheduled_query_rules_alert.
On scheduled query alert rule creation, the 'autoMitigate' property should be set to true by default.

### New or Affected Resource(s)

* azurerm_monitor_scheduled_query_rules_alert

### Potential Terraform Configuration

```hcl
# Example: Alerting Action with result count trigger
resource "azurerm_monitor_scheduled_query_rules_alert" "example" {
  name                = format("%s-queryrule", var.prefix)
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  action {
    action_group           = []
    email_subject          = "Email Header"
    custom_webhook_payload = "{}"
  }
  data_source_id = azurerm_application_insights.example.id
  description    = "Alert when total results cross threshold"
  enabled        = true
  auto_mitigate  = true
  # Count all requests with server error result code grouped into 5-minute bins
  query       = <<-QUERY
  requests
    | where tolong(resultCode) >= 500
    | summarize count() by bin(timestamp, 5m)
  QUERY
  severity    = 1
  frequency   = 5
  time_window = 30
  trigger {
    operator  = "GreaterThan"
    threshold = 3
  }
}
```

### References

* #5773

**Comment by Klaas-** [comment]
Refs https://github.com/cloutier/terraform-provider-azurerm/commit/c5aeb94dfd8c01f54894f5f07af9dfa2d657c24f

**Comment by cloutier** [comment]
Should it really be set to true by default? The default in the API is false: https://docs.microsoft.com/en-us/rest/api/monitor/scheduled-query-rules/update "The flag that indicates whether the alert should be automatically resolved or not. The default is false."

**Comment by Klaas-** [comment]
@cloutier I don't care either way -- but keeping consistent with https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_metric_alert is a valid argument, this could then be a breaking change for existing resources, right?

So maybe it's a better idea to keep it false and create another PR to set it to true for next major version :)

**Comment by cloutier** [comment]
I mean it won't be a breaking change, because it's a new property, but I agree with you anyway. I think it's better to be consistent with other terraform ressources. I'll make the change back to true in my PR.

**Comment by Klaas-** [comment]
setting it to true by default will break all alerts that use throttling -- at least that's my guess :)

**Comment by Xnnguyen** [comment]
so has the auto-mitigate been added to this module?

**Comment by Klaas-** [comment]
no, the PR is #13213

**Comment by pratikibm** [comment]
also there is limitation on frequency. if we set it more than 12hr below is the error message. 
"AutoMitigate: Stateful rules can not run in a frequency greater than 12 hours. Either reduce frequency, or set 'AutoMitigate' property to false."

**Comment by Klaas-** [comment]
Fixed by https://github.com/hashicorp/terraform-provider-azurerm/commit/2bfb63b6988700741957721084f89012a3c8f013

**Comment by Klaas-** [comment]
Released in v2.77.0 https://github.com/hashicorp/terraform-provider-azurerm/pull/13213#issuecomment-921401566 @JamesDLD feel free to close this issue

**Comment by JamesDLD** [comment]
Thanks 😊
